internal/services/groups/groups_data_source.go

```diff
@@ -156,3 +178,29 @@ func groupsDataSourceRead(ctx context.Context, d *schema.ResourceData, meta inte
 
 	return nil
 }
+
+// Extracting the flag functionality to prevent duplication
+func filterResults(securityEnabled, mailEnabled bool, results *[]msgraph.Group) (groups []msgraph.Group) {
+	if securityEnabled && !mailEnabled {
```

Comment:
We'll need a nil check for `results` here or we'll get a crash when dereferencing